sphinx/pycode/ast.py

```diff
@@ -202,6 +202,8 @@ def is_simple_tuple(value: ast.AST) -> bool:
             return "%s[%s]" % (self.visit(node.value), self.visit(node.slice))
 
     def visit_UnaryOp(self, node: ast.UnaryOp) -> str:
+        if not isinstance(node.op, ast.Not):
+            return "%s%s" % (self.visit(node.op), self.visit(node.operand))
         return "%s %s" % (self.visit(node.op), self.visit(node.operand))
```

Comment:
`ast.UnaryOp` is one of `UAdd`, `USub`, `Invert` or `Not` per [the docs](https://docs.python.org/3/library/ast.html#ast.UnaryOp) -- for `+`, `-`, and `~` we don't want a space, but for `not` we do.

A

Reply:
If it was worthwhile adding a comment here in github, maybe it is worthwhile adding the comment to the source code?

Reply:
Good suggestion, added.